models/note.go

```diff
@@ -22,16 +23,33 @@ var categoryStrings = [...]string{
 	"predictions",
 }
 
-func (category Category) String() string {
-	if category < MARGINALIA || category > PREDICTIONS {
-		return "Unknown"
+var UnDeserializeableCategoryStringError = errors.New("String does not correspond to a Note Category")
+
+func DeserializeCategory(input string) (Category, error) {
+	for i := 0; i < len(categoryStrings); i++ {
+		if input == categoryStrings[i] {
+			return Category(i), nil
+		}
 	}
+	return MARGINALIA, UnDeserializeableCategoryStringError
+}
 
+func (category Category) String() string {
 	return categoryStrings[category]
 }
 
 type Note struct {
+	Id           int64     `json:"id"`
 	AuthorId     UserId    `json:"authorId"`
 	Content      string    `json:"content"`
 	CreationTime time.Time `json:"creationTime"`
 }
+
+func CreateNewNote(userId UserId, content string) *Note {
+	return &Note{
+		Id:           -1,
```

Comment:
This is why it's weird to have `Id` on the note model. A note can exist before it has an Id.